Add bar chart for HGG vs. LGG showing all celltypes

allyhawkins opened the pull request:
Closes #180 

Here I am adding the code and the plot for Figure 5B which shows the stacked bar chart for all HGG and LGG samples that are not-multiplexed. Each bar is filled with the broad celltype group and matches up with the same colors used in the dot plot. 

- I created a new TSV file `brain-classifications.tsv` that takes all the diagnoses from the "Brain and CNS" diagnoses group and classifies them as either HGG, LGG, or other, if they don't fall into the glioma category. Note that there are a handful of diagnoses that correspond only to multiplexed samples so ultimately don't get considered, so I ended up just putting `NA` for those diagnoses groups. I also decided that since we are focusing on HGG vs LGG, we don't want to show any non-gliomas in the plot. For those and the normal I labeled as other, but I guess I could just label any samples that won't get included in the plot as NA. 
- I made two functions that I hope will be useful when working on #181 and #182. 
  - The first function just preps the consensus cell types for plotting by calculating the % of each library annotated with a given cell type. It also orders the data frame so that the libraries are ordered by total % of the library that is annotated. Note that I did not end up using `duckplyr` here since I was having issues with some of the functions that I had used when creating this data frame in OpenScPCA and it didn't feel like the time to fight with it just yet. I think since we are not looking at gene expression and just looking at cell type assignments we won't run into the same problems we did with the dot plots. 
  - The second function actually creates the bar plot, with optional faceting. For #181 we won't do any faceting, so it seemed useful to have that option here. 
- Then I added a script that creates the plot specifically for the HGG vs. LGG comparison using those two functions and saves it to a PDF. 

Below is a screenshot of the plot for easy review. I ended up including the library IDs as labels, but I'm not so sure they will remain legible once we get into 

<img width="1270" alt="Screenshot 2025-03-13 at 3 25 46 PM" src="https://github.com/user-attachments/assets/6cdde721-52ce-4139-9d1f-3fbc9f1e8eae" />

Here's an option without the library IDs. 
<img width="888" alt="Screenshot 2025-03-13 at 3 51 10 PM" src="https://github.com/user-attachments/assets/82f3bd9e-917f-46ea-9468-04fde0d46094" />

jaclyn-taroni commented on sample-info/brain-classifications.tsv:
```diff
@@ -0,0 +1,29 @@
+MONDO:0016700	Anaplastic ependymoma	Anaplastic ependymoma	Brain and CNS	NA
+MONDO:0016734	Anaplastic ganglioglioma	Anaplastic ganglioglioma	Brain and CNS	NA
+MONDO:0021640	Grade III glioma	Anaplastic glioma	Brain and CNS	High-grade glioma
+MONDO:0020560	Atypical teratoid rhabdoid tumor	Atypical teratoid rhabdoid tumor	Brain and CNS	High-grade glioma
```
ATRT are embryonal tumors: https://pmc.ncbi.nlm.nih.gov/articles/PMC8328013/#T1

jaclyn-taroni commented on sample-info/brain-classifications.tsv:
```diff
@@ -0,0 +1,29 @@
+MONDO:0020560	Atypical teratoid rhabdoid tumor	Atypical teratoid rhabdoid tumor	Brain and CNS	High-grade glioma
+MONDO:0022965	Desmoplastic infantile ganglioglioma	Desmoplastic ganglioglioma	Brain and CNS	NA
+MONDO:0005499	Brain glioma	Diffuse midline glioma	Brain and CNS	High-grade glioma
+MONDO:0005505	Dysembryoplastic neuroepithelial tumor	Dysembryoplastic neuroepithelial tumor	Brain and CNS	Low-grade glioma
```
If a ganglioglioma is not a low-grade glioma, neither is this. Both are glioneuronal and neuronal tumors: https://pmc.ncbi.nlm.nih.gov/articles/PMC8328013/#T1

jaclyn-taroni commented on sample-info/brain-classifications.tsv:
```diff
@@ -0,0 +1,29 @@
+MONDO:0022965	Desmoplastic infantile ganglioglioma	Desmoplastic ganglioglioma	Brain and CNS	NA
+MONDO:0005499	Brain glioma	Diffuse midline glioma	Brain and CNS	High-grade glioma
+MONDO:0005505	Dysembryoplastic neuroepithelial tumor	Dysembryoplastic neuroepithelial tumor	Brain and CNS	Low-grade glioma
+MONDO:0019002	Lhermitte-Duclos disease	Dysplastic gangliocytoma	Brain and CNS	Low-grade glioma
```
Same comment about ganglioglioma.

jaclyn-taroni commented:
I need more information about how you did the classifications, @allyhawkins. I would naively expect this table to be our guide: https://pmc.ncbi.nlm.nih.gov/articles/PMC8328013/#T1 I think this is harder to review with using `NA` for multiplexed diagnoses because I can't follow your logic for classification as well.

jaclyn-taroni reviewed (commented):
I am going to return my initial comments about the classifications so you can see why I think any `NA` are going to be a problem for review.

allyhawkins commented:
> I need more information about how you did the classifications, @allyhawkins. I would naively expect this table to be our guide: https://pmc.ncbi.nlm.nih.gov/articles/PMC8328013/#T1 I think this is harder to review with using `NA` for multiplexed diagnoses because I can't follow your logic for classification as well. **Edited to add:** Another approach would be to drop these from the table entirely.

I think I just googled things and totally forgot that this table existed. But I've updated this to hopefully make things more clear and better document how we make these assignments. 

- I added a small script that creates the brain classifications table. This only includes diagnoses for samples that are non-multiplexed that are in the brain and CNS projects. 
- In that script I have a list of HGG vs LGG tumor types where anything in SCPCP000001 is HGG and anything in SCPCP000002 is LGG. All other types are assigned using the table that you sent. I classified any glioneuronal tumors as LGG since ganglioglioma is part of SCPCP000002, which is LGG. ATRT and the focal cortical dysplasia are now labeled as "other" since they are neither HGG nor LGG. 
- I updated the plot based on these changes.

jaclyn-taroni commented on scripts/utils/consensus-celltype-plotting-functions.R:
```diff
@@ -189,3 +189,125 @@ marker_gene_dotplot <- function(
+    dplyr::left_join(totals_df, by = "library_id") |> 
+    dplyr::group_by(project_id, library_id, sample_id, broad_celltype_group) |> 
+    dplyr::summarize(total_cells_per_annotation = dplyr::n(),
+                     total_cells_per_library = unique(total_cells_per_library),
```
The `unique()` is required because the same total number of cells will get repeated per annotation, correct?

allyhawkins replied:
Yes that is correct.

jaclyn-taroni reviewed (approved):
Looks good! Some minor comments, but I don't necessarily need to see this again.

allyhawkins commented on scripts/utils/consensus-celltype-plotting-functions.R:
```diff
@@ -189,3 +189,125 @@ marker_gene_dotplot <- function(
+stacked_barchart <- function(
+    df, 
+    fill_column,
+    celltype_colors, # named vector where names match the values in fill_column
```
Done in https://github.com/AlexsLemonade/scpca-paper-figures/pull/208/commits/255191d0e66db079f3ee3957c307817cf9e854f2
